src/main/asciidoc/reference/elasticsearch-operations.adoc

```diff
@@ -0,0 +1,140 @@
+	public ElasticsearchTemplate elasticsearchTemplate() throws UnknownHostException { <2>
+		return new ElasticsearchTemplate(elasticsearchClient(), entityMapper());
+	}
+
+	// use the ElasticsearchEntityMapper
+	@Bean
+	@Override
+	public EntityMapper entityMapper() {                                               <3>
+		ElasticsearchEntityMapper entityMapper = new ElasticsearchEntityMapper(elasticsearchMappingContext(),
+				new DefaultConversionService());
+		entityMapper.setConversions(elasticsearchCustomConversions());
+
+		return entityMapper;
+	}
+}
+----
+
+<1> Setting up the <<elasticsearch.clients.transport>>.
+<2> Creating the `ElasticsearchTemplate` bean, offering both names, _elasticsearchOperations_ and _elasticsearchTemplate_.
+<3> Using the <<elasticsearch.mapping.meta-model>> ElasticsearchMapper.
+====
+
+
+== ElasticsearchRestTemplate
+
+The `ElasticsearchRestTemplate` is an implementation of the `ElasticsearchOperations` interface using the <<elasticsearch.clients.rest>>.
+
+.ElasticsearchRestTemplate configuration
+====
+[source,java]
+----
+@Configuration
+public class RestClientConfig extends AbstractElasticsearchConfiguration {
+	@Override
+	public RestHighLevelClient elasticsearchClient() {       <1>
+		return RestClients.create(ClientConfiguration.localhost()).rest();
+	}
+
+	// no special bean creation needed                       <2>
+
+	// use the ElasticsearchEntityMapper
+	@Bean
+	@Override
+	public EntityMapper entityMapper() {                     <3>
+		ElasticsearchEntityMapper entityMapper = new ElasticsearchEntityMapper(elasticsearchMappingContext(),
+				new DefaultConversionService());
+		entityMapper.setConversions(elasticsearchCustomConversions());
+
+		return entityMapper;
+	}
+}
+----
+<1> Setting up the <<elasticsearch.clients.rest>>.
+<2> The base class `AbstractElasticsearchConfiguration` already provides the `elasticsearchTemplate` bean.
+<3> Using the <<elasticsearch.mapping.meta-model>> ElasticsearchMapper.
+
+====
+
+== Usage examples
+
+As both `ElasticsearchTemplate` and `ElasticsearchRestTemplate` implement the `ElasticsearchOperations` interface, the code to use them is not different. The example shows how to use an injected `ElasticsearchOperations` instance in a Spring REST controller. The decision, if this is using the `TransportClient` or the `RestClient` is made by providing the
+ corresponding Bean with one of the configurations shown above.
+
+.ElasticsearchOperations usage
+====
+[source,java]
+----
+@RestController
+@RequestMapping("/")
+public class TestController {
+
+	private final ElasticsearchOperations elasticsearchOperations;
+
+	public TestController(ElasticsearchOperations elasticsearchOperations) {     <1>
+		this.elasticsearchOperations = elasticsearchOperations;
+	}
+
+	@PostMapping("/person")
+	public String save(@RequestBody Person person) {                             <2>
+
+        final IndexQuery indexQuery = new IndexQueryBuilder()
```

Comment:
There's something odd with indentation. We should also remove `final` from local variable declarations.

Reply:
done